dataline-scheduler/src/test/java/io/dataline/scheduler/DefaultSchedulerPersistenceTest.java

```diff
@@ -0,0 +1,339 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Dataline
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.scheduler;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import io.dataline.commons.json.Jsons;
+import io.dataline.config.Column;
+import io.dataline.config.DataType;
+import io.dataline.config.DestinationConnectionImplementation;
+import io.dataline.config.JobCheckConnectionConfig;
+import io.dataline.config.JobConfig;
+import io.dataline.config.JobDiscoverSchemaConfig;
+import io.dataline.config.JobSyncConfig;
+import io.dataline.config.Schema;
+import io.dataline.config.SourceConnectionImplementation;
+import io.dataline.config.StandardSync;
+import io.dataline.config.Table;
+import io.dataline.db.DatabaseHelper;
+import io.dataline.integrations.Integrations;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.jooq.Record;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.utility.MountableFile;
+
+class DefaultSchedulerPersistenceTest {
+
+  @SuppressWarnings("rawtypes")
+  private static PostgreSQLContainer container;
+  private static BasicDataSource connectionPool;
+
+  private static final SourceConnectionImplementation sourceConnectionImplementation;
```

Comment:
@michel-tricot - for the static test objects do you prefer the variable names to be camel cased or constantized e.g. `SOURCE_CONNECTION_IMPLEMENTATION`?

Reply:
`SOURCE_CONNECTION_IMPLEMENTATION`

Reply:
Also if some are only used for one test (and there are multiple tests in that testfile), you can create them in the test function directly